packages/next/server/hot-reloader.js

```diff
@@ -212,7 +212,8 @@ export default class HotReloader {
 
     const pages = createPagesMapping(
       pagePaths.filter(i => i !== null),
-      this.config.pageExtensions
+      this.config.pageExtensions,
+      false
     )
```

Comment:
After adding the test, it looks like this doesn't trigger for development since it only includes the `_app` and `_document` pages here and the rest are added in `on-demand-entry-handler` as requested so the logic for the warning might need to go further down

Reply:
It needs to be covered there indeed.